### PR TITLE
[EFR32]Enable LCD display for BRD4180 and print URL to generated QRCode for boards without LCD

### LINKED ISSUE #4188: EFR MG21 cannot use the Dev kit LCD for the QR code
### Problem
Due to pin conflicts the EFR32 MG21 cannot use the LCD on the WSTK to print the QR code as it is done with the EFR32 MG12 familily.

This breaks the Android app capacity to find the right device throught the BLE advertisement and there for prevents the network provisionning

### Proposed Solution
Implement a workaround so the Chip-tool (andoird app) can find and connect to the device through ble and do the provisionning
options :

Print the QR on terminal
Print a code that can be used to generate the QR code online.

Add info about this divergence in the example README

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
#### Problem
All device need a way to provide the user with a scannable QR code for the commisionning with chip-tool app.
A feature conflict was preventing the use of the LCD display on BRD4180 (MG21). That confict was fixed in previous fixes to MG21 code. 

The Thunderboard Sense 2 (BRD4166A) doesn't have a LCD.

 #### Summary of Changes
- Enable LCD display for BRD4180
- Print a URL to the printed QR code for the example at init for boards that do not support the LCD display
- Update readme for both apps

 Fixes #4188

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

File: examples/common/chip-app-server/QRCodeUtil.cpp [hunk 1]
```diff
@@ -35,16 +35,16 @@ void PrintQRCode(chip::RendezvousInformationFlags rendezvousFlags)
...
     {
-        ChipLogProgress(AppServer, "SetupPINCode: [%" PRIu32 "]", setupPinCode);
-        ChipLogProgress(AppServer, "SetupQRCode:  [%s]", QRCode.c_str());
-
         chip::Platform::ScopedMemoryBuffer<char> qrCodeBuffer;
         const size_t qrCodeBufferMaxSize = 3 * QRCode.size() + 1;
         qrCodeBuffer.Alloc(qrCodeBufferMaxSize);
+
+        ChipLogProgress(AppServer, "SetupPINCode: [%" PRIu32 "]", setupPinCode);
+        ChipLogProgress(AppServer, "SetupQRCode:  [%s]", QRCode.c_str());
```
Comment (bzbarsky-apple) [inline examples/common/chip-app-server/QRCodeUtil.cpp:43]: Why move these two log lines?

File: examples/common/chip-app-server/QRCodeUtil.cpp [hunk 2]
```diff
@@ -35,16 +35,16 @@ void PrintQRCode(chip::RendezvousInformationFlags rendezvousFlags)
...
         if (EncodeQRCodeToUrl(QRCode.c_str(), QRCode.size(), &qrCodeBuffer[0], qrCodeBufferMaxSize) == CHIP_NO_ERROR)
         {
-            ChipLogProgress(AppServer, "Copy/paste the below URL in a browser to see the QR Code:\n\t%s?data=%s", qrCodeBaseUrl,
-                            &qrCodeBuffer[0]);
+            ChipLogProgress(AppServer, "Copy/paste the below URL in a browser to see the QR Code:");
+            ChipLogProgress(AppServer, "%s?data=%s", qrCodeBaseUrl, &qrCodeBuffer[0]);
```
Comment (bzbarsky-apple) [inline examples/common/chip-app-server/QRCodeUtil.cpp:47]: Again, why the change, which largely seems to be getting rid of the `\t` at the beginning of the URL?

File: examples/lighting-app/efr32/README.md [hunk 3]
```diff
@@ -145,18 +166,87 @@ combination with JLinkRTTClient as follows:
 
 ## Running the Complete Example
 
--   Once the example is flashed on the board, you should be able to establish a
-    connection with an OpenThread border router. See
+-   It is assumed here that you already have a OpenThread border router
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:169]:
```suggestion
-   It is assumed here that you already have an OpenThread border router
```

File: examples/lighting-app/efr32/README.md [hunk 4]
```diff
@@ -145,18 +166,87 @@ combination with JLinkRTTClient as follows:
...
+    configured and running. If not See the following guide
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:170]:
```suggestion
    configured and running. If not, see the following guide
```

File: examples/lock-app/efr32/README.md [hunk 5]
```diff
@@ -145,18 +165,88 @@ combination with JLinkRTTClient as follows:
 
 ## Running the Complete Example
 
--   Once the example is flashed on the board, you should be able to establish a
-    connection with an OpenThread border router. See
+-   It is assumed here that you already have a OpenThread border router
```
Comment (bzbarsky-apple) [inline examples/lock-app/efr32/README.md:168]:
```suggestion
-   It is assumed here that you already have an OpenThread border router
```

File: examples/lock-app/efr32/README.md [hunk 6]
```diff
@@ -145,18 +165,88 @@ combination with JLinkRTTClient as follows:
...
+    configured and running. If not See the following guide
```
Comment (bzbarsky-apple) [inline examples/lock-app/efr32/README.md:169]:
```suggestion
    configured and running. If not, see the following guide
```

File: examples/lock-app/efr32/README.md [hunk 7]
```diff
@@ -145,18 +165,88 @@ combination with JLinkRTTClient as follows:
...
+
+        -   Pressed and hold for 6 s: Initiates the factory reset of the device.
+            Releasing the button within the 6-second window cancels the factory reset
+            procedure. **LEDs** blink in unison when the factory reset procedure is
+            initiated.
+
+    **Push Button 1**
+        Toggles the bolt state On/Off
+
+-   Once the device is provisonned and it will join the Thread network is
```
Comment (bzbarsky-apple) [inline examples/lock-app/efr32/README.md:222]:
```suggestion
-   Once the device is provisoned and it will join the Thread network is
```

File: examples/lock-app/efr32/README.md [hunk 8]
```diff
@@ -145,18 +165,88 @@ combination with JLinkRTTClient as follows:
...
+    (you can verify that the device is on the thread netowrk with the command
     `router table` using a serial terminal (screen / minicom etc.) on the board
-    running the lock-app example)
+    running the lighting-app example You can also get the address list with the
+    command ipaddr again in the serial termianl )
+
 -   Using chip-tool you can now control the lock status with on/off command such
-    as `chip-tool onoff on 1 <ipv6 address of the node> 11097`
+    as `chip-tool onoff on 1` \*\* Currently, chip-tool for Mac or Linux do not
+    yet have the Thread provisionning feature
```
Comment (bzbarsky-apple) [inline examples/lock-app/efr32/README.md:246]:
```suggestion
    yet have the Thread provisioning feature
```

File: examples/lighting-app/efr32/README.md [hunk 9]
```diff
@@ -145,18 +166,87 @@ combination with JLinkRTTClient as follows:
...
+    (you can verify that the device is on the thread netowrk with the command
     `router table` using a serial terminal (screen / minicom etc.) on the board
-    running the lighting-app example)
+    running the lighting-app example You can also get the address list with the
+    command ipaddr again in the serial termianl )
+
 -   Using chip-tool you can now control the light status with on/off command
-    such as `chip-tool onoff on 1 <ipv6 address of the node> 11097`
+    such as `chip-tool onoff on 1` \*\* Currently, chip-tool for Mac or Linux do
+    not yet have the Thread provisionning feature
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:246]:
```suggestion
    not yet have the Thread provisioning feature
```

File: examples/lighting-app/efr32/README.md [hunk 10]
```diff
@@ -145,18 +166,87 @@ combination with JLinkRTTClient as follows:
...
+
+        -   Pressed and hold for 6 s: Initiates the factory reset of the device.
+            Releasing the button within the 6-second window cancels the factory reset
+            procedure. **LEDs** blink in unison when the factory reset procedure is
+            initiated.
+
+    **Push Button 1**
+        Toggles the light state On/Off
+
+-   Once the device is provisonned and it will join the Thread network is
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:222]:
```suggestion
-   Once the device is provisioned and it will join the Thread network is
```
I can't make sense of the rest of the sentence though.  Once a device is provisioned and it will join the Thread network... then what?  Does "is established" mean to be "it established"?  Or something else?

File: examples/lighting-app/efr32/README.md [hunk 11]
```diff
@@ -145,18 +166,87 @@ combination with JLinkRTTClient as follows:
...
+    established Look for the RTT log [DL] Device Role: CHILD [DL] Partition Id:
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:223]:
```suggestion
    established, look for the RTT log [DL] Device Role: CHILD [DL] Partition Id:
```

File: examples/lighting-app/efr32/README.md [hunk 12]
```diff
@@ -145,18 +166,87 @@ combination with JLinkRTTClient as follows:
...
+    [DL] 2001:DB8::E1A2:87F1:7D5D:FECA/64 valid preferred [DL]
+    FDDE:AD00:BEEF::FF:FE00:2402/64 valid preferred rloc [DL]
+    FDDE:AD00:BEEF:0:383F:5E81:A05A:B168/64 valid preferred [DL]
+    FE80::D8F2:592E:C109:CF00/64 valid preferred [DL] LwIP Thread interface
+    addresses updated [DL] FE80::D8F2:592E:C109:CF00 IPv6 link-local address,
+    preferred) [DL] FDDE:AD00:BEEF:0:383F:5E81:A05A:B168 Thread mesh-local
+    address, preferred) [DL] 2001:DB8::E1A2:87F1:7D5D:FECA IPv6 global unicast
+    address, preferred)
+
+    Keep The gloabl unicast address, It is to be used to reach the Device with
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:235]:
```suggestion
    Keep The global unicast address; it is to be used to reach the Device with
```

File: examples/lighting-app/efr32/README.md [hunk 13]
```diff
@@ -145,18 +166,87 @@ combination with JLinkRTTClient as follows:
...
+    the chip-tool The device will be promoted to Router shortly after [DL]
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:236]:
```suggestion
    the chip-tool. The device will be promoted to Router shortly after [DL]
```

File: examples/lighting-app/efr32/README.md [hunk 14]
```diff
@@ -145,18 +166,87 @@ combination with JLinkRTTClient as follows:
...
+    Device Role: ROUTER
+
+    (you can verify that the device is on the thread netowrk with the command
     `router table` using a serial terminal (screen / minicom etc.) on the board
-    running the lighting-app example)
+    running the lighting-app example You can also get the address list with the
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:241]:
```suggestion
    running the lighting-app example. You can also get the address list with the
```

File: examples/lighting-app/efr32/README.md [hunk 15]
```diff
@@ -145,18 +166,87 @@ combination with JLinkRTTClient as follows:
...
+    command ipaddr again in the serial termianl )
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:242]:
```suggestion
    command ipaddr again in the serial terminal.)
```

File: examples/lock-app/efr32/README.md [hunk 16]
```diff
@@ -145,18 +165,88 @@ combination with JLinkRTTClient as follows:
...
+        -   _Solid On_ ; The device is fully provisioned and has full Thread
+            network and service connectivity.
+
+    **LED 1** Simulates the Lock The following states are possible:
+
+        -   _Solid On_ ; Bolt is locked
+        -   _Blinking_ ; Bolt is moving to the desired state
+        -   _Off_ ; Bolt is unlocked
+
+    **Push Button 0** - Press and Release : If not commissionned, start thread
```
Comment (bzbarsky-apple) [inline examples/lock-app/efr32/README.md:210]:
```suggestion
    **Push Button 0** - Press and Release : If not commissioned, start thread
```

File: examples/lighting-app/efr32/README.md [hunk 17]
```diff
@@ -145,18 +166,87 @@ combination with JLinkRTTClient as follows:
...
+
+        -   _Solid On_ ; The device is fully provisioned and has full Thread
+            network and service connectivity.
+
+    **LED 1** Simulates the Light The following states are possible:
+
+        -   _Solid On_ ; Light is on
+        -   _Off_ ; Light is off
+
+    **Push Button 0** - Press and Release : If not commissionned, start thread
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:210]:
```suggestion
    **Push Button 0** - Press and Release : If not commissioned, start thread
```

File: examples/lighting-app/efr32/README.md [hunk 18]
```diff
@@ -20,8 +20,16 @@ An example showing the use of CHIP on the Silicon Labs EFR32 MG12.
 ## Introduction
 
 The EFR32 lighting example provides a baseline demonstration of a Light control
-device, built using CHIP and the Silicon Labs gecko SDK. The example currently
-support OpenThread. The BLE feature is still a work in progress.
+device, built using CHIP and the Silicon Labs gecko SDK. It can be controlled by
+a Chip controller over Openthread network..
+
+The EFR32 device can be comissionned over Bluetooth Low Energy where the device
```
Comment (bzbarsky-apple) [inline examples/lighting-app/efr32/README.md:26]:
```suggestion
The EFR32 device can be commissioned over Bluetooth Low Energy where the device
```

File: examples/lock-app/efr32/README.md [hunk 19]
```diff
@@ -19,13 +19,21 @@ An example showing the use of CHIP on the Silicon Labs EFR32 MG12.
...
-built using CHIP and the Silicon Labs gecko SDK. The example currently support
-OpenThread. The BLE feature is still a work in progress.
+The EFR32 lock example provides a baseline demonstration of a door lock control
+device, built using CHIP and the Silicon Labs gecko SDK. It can be controlled by
+a Chip controller over Openthread network..
 
-The lock example is intended to serve both as a means to explore the workings of
-CHIP as well as a template for creating real products based on the Silicon Labs
-platform.
+The EFR32 device can be comissionned over Bluetooth Low Energy where the device
```
Comment (bzbarsky-apple) [inline examples/lock-app/efr32/README.md:26]:
```suggestion
The EFR32 device can be commissioned over Bluetooth Low Energy where the device
```

File: examples/lock-app/efr32/README.md [hunk 20]
```diff
@@ -19,13 +19,21 @@ An example showing the use of CHIP on the Silicon Labs EFR32 MG12.
 
 ## Introduction
 
-The EFR32 lock example provides a baseline demonstration of a door lock device,
```
Comment (bzbarsky-apple) [inline examples/lock-app/efr32/README.md:22]: A bunch of the same typos here as in lighting-app; please fix them here as well

**Review by bzbarsky-apple — APPROVED** [review]
Approved, if there's some proofreading (and maybe a spell-checker?) that happens on the documentation changes.

File: examples/common/chip-app-server/QRCodeUtil.cpp (hunk 2 above)
Comment (jmartinez-silabs) [reply examples/common/chip-app-server/QRCodeUtil.cpp:47]: Segger printing buffer size is 150 characters, with the url on the same print it was exceeding and the URL was getting trimmed so not working. Instead of increasing the Buffer size I simply did a second print as it was already on printed on a second Line.

File: examples/common/chip-app-server/QRCodeUtil.cpp (hunk 1 above)
Comment (jmartinez-silabs) [reply examples/common/chip-app-server/QRCodeUtil.cpp:43]: Just wanted to regroup those code lines thogether closer to the other Chip log process because they are all in linked to printing QRCode information

File: examples/lock-app/efr32/README.md (hunk 20 above)
Comment (jmartinez-silabs) [reply examples/lock-app/efr32/README.md:22]: It should be good now. I spell-checked both readmes

**Review by rwalker-apple — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Review by mspang — APPROVED** [review]
